### DISCUSSION TITLE
Pull Request Integration

### DISCUSSION THREAD

**Discussion by riglar** [devicecloud-dev/docs#8] [Ideas]
Inspired by Maestro Cloud's [equivalent functionality](https://cloud.mobile.dev/getting-started/pull-request-integration
). 

Mentioned in Discord [here](https://discord.com/channels/1206536148650950736/1206536149254799422/1308504899729756181).

**Comment by HugoGresse** [comment]
This is so important! 

I trigger the test using a keyword on the PR. 

then run this: 


            -   name: Run tests on Maestro CLoud
                env:
                    M_CLOUD_API_KEY: ${{ secrets.M_CLOUD_API_KEY }}
                    COMMIT_SHA: ${{ fromJson(needs.ctx.outputs.ctx).pr.merge_commit_sha }}
                    BRANCH: ${{ fromJson(needs.ctx.outputs.ctx).branch }}
                    PULL_REQUEST_ID: ${{ fromJson(needs.ctx.outputs.ctx).prNumber }}
                run: |
                    export PATH="$PATH":"$HOME/.maestro/bin"
                    iosAppPath="exported_artefacts/PlantNet.app"
                    maestro cloud \
                        -e API_URL=https://some.ngrok-free.app \
                        --apiKey "$M_CLOUD_API_KEY" \
                        --branch "$BRANCH" \
                        --repoOwner "plantnet" \
                        --repoName "plantnet-mobile" \
                        --pullRequestId "$PULL_REQUEST_ID" \
                        --commitSha "$COMMIT_SHA" \
                        --timeout 120 \
                        $iosAppPath \
                        e2e/
                        
I don't see the equivalent of this with either the current CLI or the GH Action.

**Reply by riglar** [reply]
Thanks for sharing the snippet @HugoGresse - we'd also love to see this and hope to get it into dev ASAP

**Comment by finalerock44** [comment]
This is now available in CLI 4.4.0 & GHA v2.1.0!
